### PR TITLE
introduced PsnHandle property in RegistrationDescription

### PR DESCRIPTION
**Pull request by alex-zyl**
Added PsnHandle property returning Platform-specific handle to avoid explicit type conversion to retrieve the value.

### CODE REVIEW

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs [hunk 1]
```diff
@@ -8,33 +8,31 @@
...
     using System.IO;
     using System.Linq;
     using System.Runtime.Serialization;
-    using System.Security.Cryptography;
     using System.Text;
     using System.Text.RegularExpressions;
     using System.Collections.Generic;
     using System.Xml;
-    using Microsoft.Azure.NotificationHubs.Messaging;
+    using Messaging;
```
Comment (mpodwysocki) [inline src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: Please do not shorten namespace aliases

Comment (alex-zyl) [reply src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: Sorry
Reverted

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs [hunk 2]
```diff
@@ -53,7 +51,7 @@ public abstract class RegistrationDescription : EntityDescription
         /// Initializes a new instance of the <see cref="RegistrationDescription"/> class.
         /// </summary>
         /// <param name="registration">The registration.</param>
-        public RegistrationDescription(RegistrationDescription registration)
+        protected RegistrationDescription(RegistrationDescription registration)
```
Comment (brannon) [inline src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: I agree constructors on abstract classes should be `protected`, but I'd rather not change this now. It doesn't hurt to have it `public`.

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs [hunk 3]
```diff
@@ -217,6 +215,15 @@ internal string FormattedETag
 
         internal abstract RegistrationDescription Clone();
 
+        /// <summary>
+        /// Returns platform-specific Psn handle.
```
Comment (brannon) [inline src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: Can you change "Psn" to "Pns". The acronym is short for _Platform Notification System_:

> Push notifications are delivered through platform-specific infrastructures called Platform Notification Systems (PNSes).

(from https://docs.microsoft.com/en-us/azure/notification-hubs/notification-hubs-push-notification-overview#how-push-notifications-work)

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs [hunk 4]
```diff
@@ -217,6 +215,15 @@ internal string FormattedETag
...
+        /// </summary>
+        /// 
+        /// <returns>
+        /// platform-specific Psn handle.
```
Comment (brannon) [inline src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: Same comment regarding "Psn" to "Pns".

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs [hunk 5]
```diff
@@ -217,6 +215,15 @@ internal string FormattedETag
...
+        /// </returns>
+        public string PsnHandle => GetPnsHandle();
```
Comment (brannon) [inline src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: The property name should be `PnsHandle`.

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs (hunk 3 above)
Comment (alex-zyl) [reply src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: My bad,
Had PSN (PlayStation Network) in mind when adding the property :)

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs (hunk 2 above)
Comment (alex-zyl) [reply src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: Reverted

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs (hunk 4 above)
Comment (alex-zyl) [reply src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: Changed

File: src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs (hunk 5 above)
Comment (alex-zyl) [reply src/Microsoft.Azure.NotificationHubs/RegistrationDescription.cs]: Changed

**Comment by brannon** [comment]
@alex-zyl thanks for the PR. I will merge this once we release version 3.0.2, which contains an important bug fix.

**Pull request merged**